node/src/accountant/mod.rs

```diff
@@ -1435,6 +1580,317 @@ mod tests {
+        system.run();
+        let ui_gateway_recording = ui_gateway_recording_arc.lock().unwrap();
+        assert_eq!(
+            ui_gateway_recording.get_record::<NodeToUiMessage>(0),
+            &NodeToUiMessage {
+                target: MessageTarget::ClientId(1234),
+                body: UiScanResponse {}.tmb(4321),
+            }
+        );
+    }
+
+    #[test]
+    fn scan_pending_payables_request() {
+        let config = bc_from_ac_plus_earning_wallet(
+            AccountantConfig {
+                scan_intervals: ScanIntervals {
+                    payable_scan_interval: Duration::from_millis(10_000),
+                    receivable_scan_interval: Duration::from_millis(10_000),
+                    pending_payable_scan_interval: Duration::from_secs(100),
+                },
+                when_pending_too_long_sec: DEFAULT_PENDING_TOO_LONG_SEC,
+                suppress_initial_scans: true,
+                payment_thresholds: *DEFAULT_PAYMENT_THRESHOLDS,
+            },
+            make_wallet("some_wallet_address"),
+        );
+        let fingerprint = PendingPayableFingerprint {
+            rowid_opt: Some(1234),
+            timestamp: SystemTime::now(),
+            hash: Default::default(),
+            attempt_opt: Some(1),
+            amount: 1_000_000,
+            process_error: None,
+        };
+        let pending_payable_dao = PendingPayableDaoMock::default()
+            .return_all_fingerprints_result(vec![fingerprint.clone()]);
+        let subject = AccountantBuilder::default()
+            .bootstrapper_config(config)
+            .pending_payable_dao(pending_payable_dao)
+            .build();
+        let (blockchain_bridge, _, blockchain_bridge_recording_arc) = make_recorder();
+        let subject_addr = subject.start();
+        let system = System::new("test");
+        let peer_actors = peer_actors_builder()
+            .blockchain_bridge(blockchain_bridge)
+            .build();
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+        let ui_message = NodeFromUiMessage {
+            client_id: 1234,
+            body: UiScanRequest {
+                scan_type: ScanType::PendingPayables,
+            }
+            .tmb(4321),
+        };
+
+        subject_addr.try_send(ui_message).unwrap();
+
+        System::current().stop();
+        system.run();
+        let blockchain_bridge_recording = blockchain_bridge_recording_arc.lock().unwrap();
+        assert_eq!(
+            blockchain_bridge_recording.get_record::<RequestTransactionReceipts>(0),
+            &RequestTransactionReceipts {
+                pending_payable: vec![fingerprint],
+                response_skeleton_opt: Some(ResponseSkeleton {
+                    client_id: 1234,
+                    context_id: 4321
+                }),
+            }
+        );
+    }
+
+    #[test]
+    fn report_transaction_receipts_with_response_skeleton_sends_scan_response_to_ui_gateway() {
```

Comment:
Probably even here. 

I would really like to see that changes because this file is already too long.